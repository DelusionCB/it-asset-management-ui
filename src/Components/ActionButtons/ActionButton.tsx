import React from 'react'
import {Button} from 'reactstrap';
import {useTranslation} from 'react-i18next';
import './index.scss'
import {postData} from '../../Api/Utils/archiveUtils';
import {useNavigate} from 'react-router-dom'
import {actionButtonPropTypes} from '../../Types/types.components';

function ActionButton ({values, action, type, mode, disabled = false}: actionButtonPropTypes): JSX.Element {
    const {t} = useTranslation()
    const navigate = useNavigate()

    function handleAction (): void {
        switch (action) {
            case 'create':
            case 'save':
                void postData(navigate, action, type, values)
                break;
            case 'edit':
                navigate(`/archive/${action}/${values.base_id}`, {
                    state: {
                        api: type,
                        id: values.base_id,
                        action,
                    },
                });
                window.scrollTo({top: 0, behavior: 'smooth'});
                break;
            case 'cancel':
                navigate(-1)
                window.scrollTo({top: 0, behavior: 'smooth'});
                break;
        }
    }

    return (
        <Button
            className='action-button'
            onClick={() => { handleAction(); }}
            disabled={disabled}
        >
            {t(`button.${action}`)}
        </Button>
    )
}

export default ActionButton
