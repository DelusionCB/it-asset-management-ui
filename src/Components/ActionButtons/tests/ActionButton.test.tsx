import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ActionButton from '../ActionButton';
import '@testing-library/jest-dom'
import {BrowserRouter} from 'react-router-dom';

describe('<ActionButton />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <ActionButton disabled={false} mode='create' action='create' type='application' values={{}}/>,
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
