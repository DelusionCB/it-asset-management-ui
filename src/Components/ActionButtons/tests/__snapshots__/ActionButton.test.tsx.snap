// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton /> Snapshot renders correctly 1`] = `
[
  <button
    className="action-button btn btn-secondary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    button.create
  </button>,
  ",",
]
`;
